### PR TITLE
feat: add `fix` to datalog engine and lowering

### PR DESCRIPTION
**Pull request by JonathanStarup**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Lowering.scala [hunk 1]
```diff
@@ -1009,6 +1012,16 @@ object Lowering {
       mkTag(Enums.Polarity, "Negative", innerExp, Types.Polarity, loc)
   }
 
+  private def mkFixity(f: Ast.Fixity, loc: SourceLocation): Expression = f match {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Lowering.scala:1015]: /**
    * Constructs a `Fixpoint/Ast.Fixity` from the given fixity `f`.
*/

File: main/src/library/Fixpoint/Ast/BodyPredicate.flix [hunk 2]
```diff
@@ -30,15 +30,19 @@ namespace Fixpoint/Ast {
         pub def toString(body: BodyPredicate[v]): String =
             use Fixpoint.commaSeparate;
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Ast/BodyPredicate.flix:31]: Can we match on polarity and fixity separately and just build a string that way? So we avoid the exponential combination.

File: main/src/library/Fixpoint/Stratifier.flix [hunk 3]
```diff
@@ -87,10 +87,10 @@ namespace Fixpoint {
     }
 
     def mkDepEdge(dst: HeadPredicate[v], src: BodyPredicate[v]): PrecedenceGraph = match (dst, src) {
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Stratifier.flix:89]: I know there was no comments here already, but add some comments to explain what is happening :)

**Comment by JonathanStarup** [comment]
@magnus-madsen should be good now

**Pull request merged**